pkg/containerinstall/podman.go

```diff
@@ -0,0 +1,76 @@
+package containerinstall
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"os"
+
+	"github.com/containers/podman/v4/pkg/bindings"
+	"github.com/containers/podman/v4/pkg/bindings/containers"
+	"github.com/containers/podman/v4/pkg/bindings/images"
+	"github.com/containers/podman/v4/pkg/specgen"
+	"github.com/sirupsen/logrus"
+)
+
+func getConnection(ctx context.Context, isDevelopment bool) (context.Context, error) {
+	socket := os.Getenv("ARO_PODMAN_SOCKET")
+
+	if socket == "" && isDevelopment {
+		sock_dir := os.Getenv("XDG_RUNTIME_DIR")
+		socket = "unix:" + sock_dir + "/podman/podman.sock"
+	}
+
+	if socket == "" {
+		return nil, errors.New("no podman socket defined")
+	}
+
+	return bindings.NewConnection(ctx, socket)
+}
+
+func getContainerLogs(ctx context.Context, log *logrus.Entry, containerName string) error {
+	stdout, stderr := make(chan string, 1024), make(chan string, 1024)
+	go func() {
+		for {
+			v, ok := <-stdout
+			if !ok {
+				return
+			}
+			log.Infof("stdout: %s", v)
+		}
+	}()
+
+	go func() {
+		for {
+			v, ok := <-stderr
+			if !ok {
+				return
+			}
+			log.Errorf("stderr: %s", v)
+		}
+	}()
```

Comment:
these would be better with a range over the channel instead of checking the ok value

Reply:
I copied what podman does directly, I am not super up on channels and such

Reply:
this is weird, because this is essentially reimplementing the range functionality 
```suggestion
	go func() {
		for v := range stdout {
			log.Infof("stdout: %s", v)
		}
	}()

	go func() {
		for v := range stderr {
			log.Errorf("stderr: %s", v)
		}
	}()
```